stock_move_line_removal_date/models/stock_move_line.py

```diff
@@ -37,9 +37,8 @@ def convert_date_to_datetime(self, date, hour):
         su_timestamp = pytz.timezone(su_tz).localize(dt).astimezone(pytz.utc)
         # Following conditional branching is needed due to peculiar behavior
         # of how the diff is calculated depending on the timezone.
+        diff = (su_timestamp - utc_timestamp).seconds / 3600
         if utc_timestamp >= su_timestamp:
             diff = (utc_timestamp - su_timestamp).seconds / 3600
             return utc_timestamp - relativedelta(hours=diff)
```

Comment:
```suggestion
        diff = -(utc_timestamp - su_timestamp).seconds / 3600
        if su_timestamp > utc_timestamp:
            diff = (su_timestamp - utc_timestamp).seconds / 3600
        return utc_timestamp + relativedelta(hours=diff)
```

This would be slightly better for the Japan timezone.